modules/core/src/test/java/org/apache/ignite/internal/processors/metastorage/DistributedMetaStoragePersistentTest.java

```diff
@@ -73,19 +96,174 @@
     public void testRestart() throws Exception {
         IgniteEx ignite = startGrid(0);
 
-        ignite.cluster().active(true);
+        ignite.cluster().state(ClusterState.ACTIVE);
 
         ignite.context().distributedMetastorage().write("key", "value");
 
         stopGrid(0);
 
         ignite = startGrid(0);
 
-        ignite.cluster().active(true);
+        ignite.cluster().state(ClusterState.ACTIVE);
 
         assertEquals("value", ignite.context().distributedMetastorage().read("key"));
     }
 
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testStoreLagOnOneNode() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        IgniteEx ignite2 = startGrid(1);
+
+        DistributedMetaStorageImpl distrMetaStore = (DistributedMetaStorageImpl)ignite.context().distributedMetastorage();
+
+        DmsDataWriterWorker worker = GridTestUtils.getFieldValue(distrMetaStore, "worker");
+
+        assertNotNull(worker);
+
+        // check we still have no cluster tag key.
+        assertNull(distrMetaStore.read(CLUSTER_ID_TAG_KEY));
+
+        GridTestUtils.setFieldValue(worker, "writeCondition", new Predicate<String>() {
+            private volatile boolean skip;
+
+            @Override public boolean test(String s) {
+                if (s.equals(CLUSTER_ID_TAG_KEY) || skip) {
+                    skip = true;
+
+                    return true;
+                }
+                return false;
+            }
+        });
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        String clusterTag = "griffon";
+
+        assertTrue(GridTestUtils.waitForCondition(() -> {
+            boolean fail = false;
```

Comment:
This can be replaces with `return true;` in `try` and `return false;` in `catch`.